src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -837,7 +837,7 @@ private static FieldSymbol CheckIsTupleElement(SyntaxNode node, NamedTypeSymbol
 
         private BoundPattern BindVarPattern(VarPatternSyntax node, TypeSymbol inputType, uint inputValEscape, bool hasErrors, DiagnosticBag diagnostics)
         {
-            if (inputType.IsPointerType() &&
+            if (inputType.IsPointerOrFunctionPointer() &&
                 (node.Designation.Kind() == SyntaxKind.ParenthesizedVariableDesignation ||
```

Comment:
>node.Designation.Kind() == SyntaxKind.ParenthesizedVariableDesignation [](start = 17, length = 70)

Note that this check is not language version specific. We might want to test this specially, if not already.